### DISCUSSION TITLE
negative lookahead assertion

### DISCUSSION THREAD

**Discussion by electriquo** [renovatebot/renovate#24345] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Javascript supports [negative lookahead assertion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Assertions#other_assertions) but renovate config validator fails for any such string. E.g. for the following renovate (json) configuration

```json
{
    "regexManagers": [
    {
      "fileMatch": [".*"],
      "matchStrings": [
        "q(?!u)b"
      ]
    }
  ]
}
```

renovate config validator throws configuration error

```shell
"errors": [
  {
    "topic": "Configuration Error",
    "message": "Invalid regExp for regexManagers: `q(?!u)b`"
  }
]
```

Is this a bug or is the regex string is incorrect?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
Renovate uses `re2` for regex so this is expected
